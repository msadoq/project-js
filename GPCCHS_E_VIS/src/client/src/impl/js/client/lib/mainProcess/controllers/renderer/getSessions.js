import getLogger from 'common/log';
import { server } from '../../ipc';
import { getStore } from '../../../store/mainStore';
import { updateSessions } from '../../../store/actions/sessions';

const logger = getLogger('main:controllers:renderer:getSessions');

export default function (reply, event, payload, queryId) {
  server.requestSessions((err, sessions = []) => {
    if (err) {
      return logger.error(err);
    }

    // update sessions list in store
    getStore().dispatch(updateSessions(sessions));

    // inform renderer process that sessions are up to date
    reply({ event: 'runCallback', queryId });
  });
}
