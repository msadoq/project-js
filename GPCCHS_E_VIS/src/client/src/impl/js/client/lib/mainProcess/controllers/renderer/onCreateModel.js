import _cloneDeep from 'lodash/cloneDeep';
import { dirname } from 'path';
import { getStore } from '../../../store/mainStore';
import { getView } from '../../../store/selectors/views';
import { add } from '../../../store/actions/messages';
import { getPathByFilePicker } from '../../dialog';
import { saveViewAs } from '../../../common/documentManager';
import vivl from '../../../../VIVL/main';
import { getRootDir } from '../../../common/fmd';


const root = getRootDir();

export default function ({ viewId }) {
  const { getState, dispatch } = getStore();
  const { type, configuration, absolutePath } = getView(getState(), viewId);

  const folder = absolutePath ? dirname(absolutePath) : root;
  return getPathByFilePicker(folder, 'model', 'save', (err, path) => {
    if (path) {
      // Create a twin view without formula in entry Points
      const modelConfig = _cloneDeep(configuration);
      const structureType = vivl(type, 'structureType')();
      if (structureType === 'last') {
        modelConfig.entryPoints.forEach((ep) => {
          ep.connectedData.formula = ''; // eslint-disable-line no-param-reassign
        });
      } else if (structureType === 'range') {
        modelConfig.entryPoints.forEach((ep) => {
          ep.connectedDataX.formula = ''; // eslint-disable-line no-param-reassign
          ep.connectedDataY.formula = ''; // eslint-disable-line no-param-reassign
        });
      }

      saveViewAs(modelConfig, type, path, (errSaving) => {
        if (errSaving) {
          dispatch(add(viewId, 'danger', `Model unsaved ${errSaving}`));
        } else {
          dispatch(add(viewId, 'success', 'Model saved'));
        }
      });
    }
  });
}
