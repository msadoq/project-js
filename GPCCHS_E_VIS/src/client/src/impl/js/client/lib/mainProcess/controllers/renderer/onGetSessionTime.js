import getLogger from 'common/log';
import reply from 'common/ipc/reply';
import { server } from '../../ipc';

const logger = getLogger('main:controllers:renderer:onGetSessionTime');

export default function (queryId) {
  server.requestSessionTime(({ err, time }) => {
    if (err) {
      return logger.error(err);
    }

    reply(queryId, time);
  });
}
