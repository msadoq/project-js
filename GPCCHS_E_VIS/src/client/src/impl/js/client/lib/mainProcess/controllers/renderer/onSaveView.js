import { dirname } from 'path';
import { getStore } from '../../../store/mainStore';
import { getView } from '../../../store/selectors/views';
import { setModified, updatePath, updateAbsolutePath, setViewOid } from '../../../store/actions/views';
import { addOnce as addMessage } from '../../../store/actions/messages';
import { getPathByFilePicker } from '../../dialog';
import { saveViewAs } from '../../../common/documentManager';

import { getRootDir, isFmd, getRelativeFmdPath } from '../../../common/fmd';

const getPath = path => (isFmd(path) ? getRelativeFmdPath(path) : path);

const root = getRootDir();
const addViewError = (viewId, msg) => addMessage(viewId, 'danger', msg);

export default function ({ viewId, saveAs }) {
  const { getState, dispatch } = getStore();
  const { type, configuration, absolutePath } = getView(getState(), viewId);

  function oncePath(savingAbsolutePath) {
    saveViewAs(configuration, type, savingAbsolutePath, (err, oid) => {
      if (err) {
        return getStore().dispatch(addViewError(viewId, err));
      }
      if (saveAs !== savingAbsolutePath) {
        // only for 'Save as...' action
        dispatch(updatePath(viewId, getPath(savingAbsolutePath)));
        dispatch(updateAbsolutePath(viewId, savingAbsolutePath));
        if (oid) {
          dispatch(setViewOid(viewId, oid));
        }
      }

      dispatch(setModified(viewId, false));
      dispatch(addMessage(viewId, 'success', 'View saved'));
    });
  }

  if (saveAs) {
    return oncePath(saveAs);
  }

  const folder = absolutePath ? dirname(absolutePath) : root;
  return getPathByFilePicker(folder, 'view', 'save', (err, path) => oncePath(path));
}
