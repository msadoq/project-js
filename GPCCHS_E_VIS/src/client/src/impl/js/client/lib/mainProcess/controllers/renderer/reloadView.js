/**
 * Created by isis on 1/5/17.
 */
