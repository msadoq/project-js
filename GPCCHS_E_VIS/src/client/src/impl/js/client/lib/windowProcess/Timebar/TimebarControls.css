.controlsUl{
  display: table;
  margin: 0px auto;
  padding: 0px;
}
.controlsLi{
  display: table-cell;
}
.controlButton{
  min-width: 30px;
  margin-right: 5px;
  border-radius: 1px;
  height: 19px;
  line-height: 1px;
  font-weight: 600;
  font-size: 0.8em;

}
.controlButtonPause{
  font-weight: 600;
  text-align: center;
  width: 30px;
}
.controlButtonPlay{
  width: 30px;
  background: #60a7e5;
  box-shadow: 0px 0px 6px #337ab7 inset;
  border: 1px solid #337ab7;
  font-weight: 600;
  font-size: 8px;
  padding-left: 9px;
}
.controlButtonPlay:hover{
  background: #549edd;
  box-shadow: 0px 0px 6px #337ab7 inset;
  border: 1px solid #337ab7;
}
