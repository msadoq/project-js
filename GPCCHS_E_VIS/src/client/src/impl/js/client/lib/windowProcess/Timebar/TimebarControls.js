import React, { Component } from 'react';
import classnames from 'classnames';
import { Col, Row } from 'react-bootstrap';
import globalConstants from 'common/constants';
import styles from './TimebarControls.css';
import compute from '../../mainProcess/play';

const currentUpperMargin = 1 / 100;

export default class TimebarControls extends Component {

  static propTypes = {
    timebarId: React.PropTypes.string.isRequired,
    timebarPlayingState: React.PropTypes.string.isRequired,
    timebarMode: React.PropTypes.string.isRequired,
    timebarSpeed: React.PropTypes.number.isRequired,
    visuWindow: React.PropTypes.object.isRequired,
    slideWindow: React.PropTypes.object.isRequired,
    viewport: React.PropTypes.object,
    updatePlayingState: React.PropTypes.func.isRequired,
    updateSpeed: React.PropTypes.func.isRequired,
    onChange: React.PropTypes.func.isRequired,
    updateMode: React.PropTypes.func.isRequired,
    currentSessionOffsetMs: React.PropTypes.number,
  }

  componentDidUpdate() {
    const { timebarPlayingState } = this.props;
    if (timebarPlayingState === 'play' && !this.timeout) this.tick();
    if (timebarPlayingState === 'pause' && this.timeout) {
      clearTimeout(this.timeout);
      this.timeout = null;
    }
  }

  changeSpeed = (dir) => {
    const { updateSpeed, timebarId, timebarSpeed, timebarPlayingState } = this.props;
    let newSpeed = dir === 'up' ? 2 * timebarSpeed : timebarSpeed / 2;
    if (timebarSpeed === 0.1) {
      newSpeed = 0.125;
    } else if (timebarSpeed === 10) {
      newSpeed = 8;
    }
    if (newSpeed > 10) newSpeed = 10;
    if (newSpeed < 0.1) newSpeed = 0.1;
    updateSpeed(timebarId, newSpeed);
    if (newSpeed !== 1 && timebarPlayingState === 'pause') this.togglePlayingState();
  }

  goNow = (e) => {
    e.preventDefault();
    const { onChange, timebarId } = this.props;
    const { lower, upper } = this.props.visuWindow;

    const nowMs = Date.now();
    const msWidth = upper - lower;
    const newLower = nowMs - msWidth;
    onChange(
      timebarId,
      {
        lower: newLower,
        upper: nowMs,
        current: nowMs,
        viewport: {
          lower: newLower - (2 * (nowMs - newLower)),
          upper: nowMs + ((nowMs - newLower) / 5)
        }
      }
    );
  }

  jump = (e) => {
    e.preventDefault();
    const { onChange, timebarId, slideWindow } = this.props;
    const { lower, upper, current } = this.props.visuWindow;

    const movedMs = 1000 * e.currentTarget.getAttribute('offset');
    onChange(
      timebarId,
      {
        lower: lower + movedMs,
        upper: upper + movedMs,
        current: current + movedMs,
        slideWindow: {
          lower: slideWindow.lower + movedMs,
          upper: slideWindow.upper + movedMs,
        }
      }
    );
  }

  switchMode = (e) => {
    e.preventDefault();
    const { onChange, timebarId, timebarMode, updateMode,
      visuWindow, currentSessionOffsetMs, slideWindow } = this.props;
    const { lower, upper, current } = visuWindow;
    const mode = e.currentTarget.getAttribute('mode');

    if (mode === timebarMode) return;

    // Realtime is not really a mode, we just go to session realtime and play
    if (mode === 'Realtime') {
      if (mode !== 'Normal') updateMode(timebarId, 'Normal');
      const msWidth = upper - lower;
      const realTimeMs = Date.now() + currentSessionOffsetMs;
      const newLower = realTimeMs - ((1 - currentUpperMargin) * msWidth);
      const newUpper = realTimeMs + (currentUpperMargin * msWidth);
      onChange(
        timebarId,
        {
          lower: newLower,
          upper: newUpper,
          current: realTimeMs,
          slideWindow: {
            lower: newLower - ((newUpper - newLower) * 2),
            upper: newUpper + ((newUpper - newLower) / 5)
          }
        }
      );
      this.togglePlayingState(null, 'play');
    } else {
      if (mode === 'Extensible' && slideWindow.upper < upper) {
        onChange(
          timebarId,
          {
            slideWindow: {
              lower: slideWindow.lower,
              upper: upper + ((upper - lower) / 4)
            }
          }
        );
      } else if (mode === 'Fixed' && slideWindow.upper > upper) {
        onChange(
          timebarId,
          {
            slideWindow: {
              lower: slideWindow.lower,
              upper: upper - ((upper - current) / 2)
            }
          }
        );
      }
      updateMode(timebarId, mode);
    }
  }

  togglePlayingState = (e, mode = null) => {
    const { updatePlayingState, timebarId, timebarPlayingState } = this.props;
    const newtimebarPlayingState = mode || (timebarPlayingState === 'pause' ? 'play' : 'pause');
    if (timebarPlayingState === newtimebarPlayingState) return;
    updatePlayingState(
      timebarId,
      newtimebarPlayingState
    );
    if (newtimebarPlayingState === 'play') {
      this.tick();
    } else {
      clearTimeout(this.timeout);
      this.timeout = null;
    }
  }

  tick() {
    this.timeout = setTimeout(
      () => {
        const { onChange, timebarId, timebarSpeed, timebarMode,
          viewport, slideWindow, visuWindow } = this.props;
        const { lower, upper, current } = visuWindow;

        const newCurrent = current + (globalConstants.HSC_PLAY_FREQUENCY * timebarSpeed);
        const cursors = compute(newCurrent, lower, upper, slideWindow.lower, slideWindow.upper,
          viewport.lower, viewport.upper, timebarMode, currentUpperMargin);
        onChange(
          timebarId,
          {
            lower: cursors[0],
            upper: cursors[1],
            current: newCurrent,
            viewport: { lower: cursors[4], upper: cursors[5] },
            slideWindow: { lower: cursors[2], upper: cursors[3] }
          }
        );
        this.tick();
      },
      globalConstants.HSC_PLAY_FREQUENCY
    );
  }

  render() {
    const { timebarPlayingState, timebarSpeed,
      timebarMode, currentSessionOffsetMs, viewport } = this.props;
    const opTimebarPlayingState = timebarPlayingState === 'pause' ? 'play' : 'pause';

    if (!viewport) return (<div />);

    const allButtonsKlasses = classnames('btn', 'btn-xs', 'btn-control');

    return (
      <div>
        <Col xsOffset={3} xs={9}>
          <Row>
            <Col xs={12}>
              <ul className={styles.controlsUl}>
                <li className={styles.controlsLi}>
                  <button
                    className={allButtonsKlasses}
                    onClick={this.changeSpeed}
                    title="Decrease speed"
                  >
                    &#9668;&#9668;
                  </button>
                </li>
                <li className={styles.controlsLi}>
                  <button
                    className={classnames('btn', 'btn-xs', 'btn-default')}
                  >
                    {`${timebarSpeed}X`}
                  </button>
                </li>
                <li className={styles.controlsLi}>
                  <button
                    className={allButtonsKlasses}
                    onClick={this.changeSpeed.bind(null, 'up')}
                    title="Increase speed"
                  >
                    &#9658;&#9658;
                  </button>
                </li>
                <li className={styles.controlsLi}>
                  <button
                    className={classnames(
                      allButtonsKlasses,
                      {
                        [styles.controlButtonPlay]: timebarPlayingState === 'play',
                        [styles.controlButtonActive]: timebarPlayingState === 'play',
                        [styles.controlButtonPause]: timebarPlayingState === 'pause'
                      }
                    )}
                    onClick={this.togglePlayingState}
                    title={opTimebarPlayingState}
                  >
                    {timebarPlayingState === 'play' ? <span>&#9613;&#9613;</span> : <span>&#9658;</span>}
                  </button>
                </li>
                <li className={styles.controlsLi}>
                  <button
                    className={allButtonsKlasses}
                    onClick={this.goNow}
                    title="Go now"
                  >
                    NOW
                  </button>
                </li>
                <li className={styles.controlsLi}>
                  <button
                    offset={-10}
                    className={allButtonsKlasses}
                    onClick={this.jump}
                    title="- 10s"
                  >
                    - 10s
                  </button>
                </li>
                <li className={styles.controlsLi}>
                  <button
                    offset={10}
                    className={allButtonsKlasses}
                    onClick={this.jump}
                    title="+ 10s"
                  >
                    + 10s
                  </button>
                </li>
              </ul>
              <ul className={classnames(styles.controlsUl, styles.controlsUlRight)}>
                <li className={styles.controlsLi}>
                  <button
                    mode="Normal"
                    className={classnames(allButtonsKlasses, { [styles.controlButtonActive]: (timebarMode === 'Normal') })}
                    onClick={this.switchMode}
                    title="Normal mode"
                  >
                    Normal
                  </button>
                </li>
                <li className={styles.controlsLi}>
                  <button
                    mode="Extensible"
                    className={classnames(allButtonsKlasses, { [styles.controlButtonActive]: (timebarMode === 'Extensible') })}
                    onClick={this.switchMode}
                    title="Extensible mode"
                  >
                    Extensible
                  </button>
                </li>
                <li className={styles.controlsLi}>
                  <button
                    mode="Fixed"
                    className={classnames(allButtonsKlasses, { [styles.controlButtonActive]: (timebarMode === 'Fixed') })}
                    onClick={this.switchMode}
                    title="Fixed mode"
                  >
                    Fixed
                  </button>
                </li>
                <li className={styles.controlsLi}>
                  <button
                    mode="Realtime"
                    className={classnames(allButtonsKlasses, { [styles.controlButtonActive]: (timebarMode === 'Realtime') })}
                    onClick={this.switchMode}
                    title={currentSessionOffsetMs ? 'Real time mode' : "No master track is set, can't go realtime"}
                    disabled={currentSessionOffsetMs ? '' : 'disabled'}
                  >
                    Real time
                  </button>
                </li>
              </ul>
            </Col>
          </Row>
        </Col>
      </div>
    );
  }
}
