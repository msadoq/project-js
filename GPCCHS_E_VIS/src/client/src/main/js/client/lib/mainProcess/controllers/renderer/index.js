// ====================================================================
// HISTORY
// VERSION : 1.1.0 : : : 28/02/2017 : Initial version
// VERSION : 1.1.2 : DM : #3622 : 07/03/2017 : first draft on inspector: retrieve data from rtd on right-click
// VERSION : 1.1.2 : DM : #5822 : 16/03/2017 : resolve a rtd link in the inspector
// VERSION : 1.1.2 : DM : #5828 : 30/03/2017 : Add a F1 button in VIMA to open the docu wiki helper
// VERSION : 1.1.2 : DM : #5822 : 03/04/2017 : merge dev in working branch
// VERSION : 1.1.2 : DM : #5828 : 23/05/2017 : Move common/callback and common/ipc modules in client sub-component
// VERSION : 1.1.2 : DM : #6785 : 12/06/2017 : activate links in views .
// VERSION : 1.1.2 : DM : #5828 : 13/06/2017 : Move common/constants/ in client/ folder
// VERSION : 1.1.2 : DM : #5828 : 14/06/2017 : Cleanup client/ file organization and test helper modules
// VERSION : 1.1.2 : FA : ISIS-FT-2132 : 15/06/2017 : Ask to save before closing view or page
// VERSION : 1.1.2 : DM : #6700 : 16/06/2017 : Add store enhancers helpers code coverage and merge with dev
// VERSION : 1.1.2 : DM : #6700 : 26/06/2017 : Change windowProcess health management to dispatch a Redux action directly.
// VERSION : 1.1.2 : DM : #6785 : 29/06/2017 : Fix opening view link in a new page and read only path for link definition
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : First draft on catalog explorer
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : catalog explorer : open, close and browse items
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 06/07/2017 : Remove unused onSavePage ipc renderer to mainProcess controllers
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 06/07/2017 : Replace ipc openPage by askOpenPage redux action
// VERSION : 1.1.2 : FA : #7235 : 18/07/2017 : Clean old code in menuManager/IPC controller
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 18/07/2017 : Fix common/fmd, disable Links feature for now
// VERSION : 1.1.2 : DM : #6700 : 20/07/2017 : Remove obsolete onServerDebug interface, ipcs and model
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about openInspector .
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about FMD_GET/FMD_CREATE and RELOAD_SESSIONS/GET_SESSION_TIME
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about documents .
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Remove serverDebug ipc . .
// VERSION : 1.1.2 : DM : #6700 : 03/08/2017 : Merge branch 'dev' into dbrugne-data
// VERSION : 1.1.2 : FA : #7145 : 04/08/2017 : Clean IPC about opening wiki helper + create a store folder in mainProcess
// END-HISTORY
// ====================================================================

import constants from 'constants';

import handle from 'common/ipc/handle';
import reply from 'common/ipc/reply';

import onReduxCurrentState from './onReduxCurrentState';
import onReduxDispatch from './onReduxDispatch';
import onResolveLink from './onResolveLink';
import onResolveRteLink from './onResolveRteLink';
import onGetRteCatalogs from './onGetRteCatalogs';
import onGetRteItemNames from './onGetRteItemNames';
import onOpenRteItem from './onOpenRteItem';
import onFocusRteItem from './onFocusRteItem';
import onGetRteDomains from './onGetRteDomains';

const controller = {
  [constants.IPC_METHOD_REDUX_CURRENT_STATE]: (...args) => onReduxCurrentState(reply, ...args),
  [constants.IPC_METHOD_REDUX_DISPATCH]: onReduxDispatch,
  [constants.IPC_METHOD_RESOLVE_LINK]: onResolveLink,
  [constants.IPC_METHOD_GET_RTE_DOMAINS]: onGetRteDomains,
  [constants.IPC_METHOD_GET_RTE_CATALOGS]: onGetRteCatalogs,
  [constants.IPC_METHOD_GET_RTE_ITEM_NAMES]: onGetRteItemNames,
  [constants.IPC_METHOD_OPEN_RTE_ITEM]: onOpenRteItem,
  [constants.IPC_METHOD_FOCUS_RTE_ITEM]: onFocusRteItem,
  [constants.IPC_METHOD_RESOLVE_RTE_LINK]: onResolveRteLink,
};

export default (electronEvent, data) => handle(
  controller,
  data,
  response => electronEvent.sender.send('global', response)
);
