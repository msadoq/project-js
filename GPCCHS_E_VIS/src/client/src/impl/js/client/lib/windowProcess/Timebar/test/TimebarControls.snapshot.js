import React from 'react';
import renderer from 'react-test-renderer';
import TimebarFixture from '../../../../test/__mocks__/Timebar';
import TimebarControls from '../TimebarControls';

const propsStub = {
  updateSpeed: () => null,
  updateCursors: () => null,
  updateMode: () => null,
  pause: () => null,
  play: () => null,
  isPlaying: true,
  visuWindow: TimebarFixture.visuWindow,
  slideWindow: TimebarFixture.slideWindow,
  timebarSpeed: TimebarFixture.speed,
  timebarMode: TimebarFixture.mode,
  timebarId: TimebarFixture.timebarId,
  currentSessionOffsetMs: 800,
};

test('TimebarControls renders correctly', () => {
  const tree = renderer.create(
    <TimebarControls {...propsStub} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
