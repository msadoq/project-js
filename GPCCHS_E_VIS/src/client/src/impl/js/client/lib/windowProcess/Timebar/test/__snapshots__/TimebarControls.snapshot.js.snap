exports[`test Controls renders correctly 1`] = `
<div>
  <ul
    className="controlsUl">
    
    <li
      className="controlsLi">
      <button
        className="btn btn-xs btn-control"
        onClick={[Function]}
        title="Decrease speed">
        <span
          className="glyphIcon glyphicon glyphicon-backward" />
      </button>
    </li>
    <li
      className="controlsLi">
      <button
        aria-describedby="playingSpeedPopover"
        className="btn btn-xs btn-default"
        onClick={[Function]}>
        1X
      </button>
    </li>
    <li
      className="controlsLi">
      <button
        className="btn btn-xs btn-control"
        onClick={[Function]}
        title="Increase speed">
        <span
          className="glyphIcon glyphicon glyphicon-forward" />
      </button>
    </li>
    <li
      className="controlsLi">
      <button
        className="btn btn-xs btn-control controlButtonPlay controlButtonActive"
        onClick={[Function]}
        title="pause">
        <span
          className="glyphIcon glyphicon glyphicon-pause" />
      </button>
    </li>
    <li
      className="controlsLi">
      <button
        className="btn btn-xs btn-control"
        offset={-10}
        onClick={[Function]}
        title="- 10s">
        - 10s
      </button>
    </li>
    <li
      className="controlsLi">
      <button
        className="btn btn-xs btn-control"
        offset={10}
        onClick={[Function]}
        title="+ 10s">
        + 10s
      </button>
    </li>
    <li
      className="controlsLi">
      <button
        className="btn btn-xs btn-control"
        onClick={[Function]}
        title="Go now">
        NOW
      </button>
    </li>
    <li
      className="controlsLi">
      <button
        className="btn btn-xs btn-control"
        onClick={[Function]}
        title="Restore visu window\'s default width">
        _
      </button>
    </li>
    <li
      className="controlsLi">
      <button
        className="btn btn-xs btn-control"
        onClick={[Function]}
        title="Display time setter">
        <span
          className="glyphicon glyphicon-align-justify" />
      </button>
    </li>
  </ul>
  <div>
    <ul
      className="controlsUl controlsUlRight">
      <li
        className="controlsLi">
        <button
          className="btn btn-xs btn-control controlButtonActive"
          mode="Normal"
          onClick={[Function]}
          title="Normal mode">
          Normal
        </button>
      </li>
      <li
        className="controlsLi">
        <button
          className="btn btn-xs btn-control"
          mode="Extensible"
          onClick={[Function]}
          title="Extensible mode">
          Extensible
        </button>
      </li>
      <li
        className="controlsLi">
        <button
          className="btn btn-xs btn-control"
          mode="Fixed"
          onClick={[Function]}
          title="Fixed mode">
          Fixed
        </button>
      </li>
      <li
        className="controlsLi">
        <button
          className="btn btn-xs btn-control"
          disabled={true}
          mode="Realtime"
          onClick={[Function]}
          title="No master track is set, can\'t go realtime">
          Real time
        </button>
      </li>
    </ul>
  </div>
</div>
`;
